src/api-service/__app__/onefuzzlib/azure/auto_scale.py

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env python
+#
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+from datetime import datetime, timedelta
+from enum import auto
+import logging
+from typing import Any, Dict, List, Optional, Set, Union, cast
+from uuid import UUID
+import uuid
+from onefuzztypes.enums import ErrorCode
+
+from onefuzztypes.models import Error
+from onefuzztypes.primitives import Region
+from .vmss import get_vmss_size
+from .monitor import get_monitor_client
+
+from azure.mgmt.monitor.models import (
+    AutoscaleProfile,
+    AutoscaleSettingResource,
+    ScaleCapacity,
+    ScaleRule,
+    MetricTrigger,
+    ScaleAction,
+    TimeAggregationType,
+    ComparisonOperationType,
+    MetricStatisticType,
+    ScaleDirection,
+    ScaleType,
+)
+from azure.core.exceptions import (
+    HttpResponseError,
+    ResourceExistsError,
+    ResourceNotFoundError,
+)
+from msrestazure.azure_exceptions import CloudError
+
+from .creds import (
+    get_base_region,
+    get_base_resource_group,
+    get_subscription,
+    retry_on_auth_failure,
+)
+
+# TODO: Look into deleting auto scale resource when deleting scaleset
```

Comment:
Tested this locally using `onefuzz scalesets shutdown`. Azure automatically deletes the auto scale resource so we don't need to add any additional logic for that case.